src/window_gui.h

```diff
@@ -287,8 +290,11 @@ struct Window : ZeroedMemoryAllocator {
 public:
 	Window(WindowDesc *desc);
 
+protected:
+	/* Protected to prevent deletion anywhere outside Window::DeleteClosedWindows(). */
 	virtual ~Window();
 
+public:
```

Comment:
Perhaps it would make sense to reorder some things to prevent duplicate private/protected/public declarations?